File: factorial.c
#include <stdio.h>
#include <stdlib.h>
#include <Windows.h>

unsigned long long factorial_iterativo(int n) {
    unsigned long long resultado = 1;
    for (int i = 1; i <= n; i++) {
        resultado *= i;
    }
    return resultado;
}

unsigned long long factorial_recursivo(int n) {
    if (n == 0) {
        return 1;
    }
    return n * factorial_recursivo(n - 1);
}

int main() {
    int n;
    printf("Ingrese un numero para calcular su factorial: ");
    scanf("%d", &n);

    LARGE_INTEGER frecuencia;
    LARGE_INTEGER inicio;
    LARGE_INTEGER fin;

    QueryPerformanceFrequency(&frecuencia);
    QueryPerformanceCounter(&inicio);
    unsigned long long factorial_iter = factorial_iterativo(n);
    QueryPerformanceCounter(&fin);

    double tiempo_iterativo = (double)(fin.QuadPart - inicio.QuadPart) * 1000000 / frecuencia.QuadPart;

    QueryPerformanceCounter(&inicio);
    unsigned long long factorial_rec = factorial_recursivo(n);
    QueryPerformanceCounter(&fin);

    double tiempo_recursivo = (double)(fin.QuadPart - inicio.QuadPart) * 1000000 / frecuencia.QuadPart;

    printf("Factorial calculado de manera iterativa: %llu\n", factorial_iter);
    printf("Tiempo de ejecucion en microsegundos (iterativo): %.6f\n", tiempo_iterativo);
    printf("Factorial calculado de manera recursiva: %llu\n", factorial_rec);
    printf("Tiempo de ejecucion en microsegundos (recursivo): %.6f\n", tiempo_recursivo);

    return 0;
}

File: factorial.py
import time

def factorial_iterativo(n):
    factorial = 1
    for i in range(1, n + 1):
        factorial *= i
    return factorial

def factorial_recursivo(n):
    if n == 0:
        return 1
    else:
        return n * factorial_recursivo(n - 1)

# Pedir al usuario el número para calcular su factorial
n = int(input("Ingrese un número para calcular su factorial: "))

# Calcular factorial de forma iterativa y medir tiempo
inicio_iterativo = time.time()
res_iterativo = factorial_iterativo(n)
fin_iterativo = time.time()
tiempo_iterativo = (fin_iterativo - inicio_iterativo) / 1000

# Calcular factorial de forma recursiva y medir tiempo
inicio_recursivo = time.time()
res_recursivo = factorial_recursivo(n)
fin_recursivo = time.time()
tiempo_recursivo = (fin_recursivo - inicio_recursivo) / 1000

print("Resultado del factorial de", n, "igual a: Recursivo =", res_recursivo, " Iterativo:", res_iterativo)
print(f"Factorial de {n} calculado de forma iterativa en {tiempo_iterativo:.9f} milisegundos.")
print(f"Factorial de {n} calculado de forma recursiva en {tiempo_recursivo:.9f} milisegundos.")
